fstat/controller.py

```diff
@@ -1,50 +1,87 @@
 from collections import Counter
-from flask import render_template, redirect, url_for
+from flask import render_template, redirect, url_for, request
 from fstat import app, db
-from fstat.lib import x_days_ago
 from model import Failure, FailureInstance
-from datetime import datetime
+from datetime import datetime, timedelta
 
 
 @app.route("/")
 def index():
-    return redirect(url_for('weekly_overall_summary', num=1))
+    return redirect(url_for('overall_summary'))
 
 
-@app.route("/weeks/<int:num>")
-def weekly_overall_summary(num=None):
-    if num > 12:
-        num = 12
-    cut_off_date = (x_days_ago(num*7))
-    failure_instances = FailureInstance.query.filter(
-            FailureInstance.timestamp > cut_off_date)
+@app.route('/summary')
+def overall_summary():
+    ''' 
+        Shows overall summary
+        Params
+            - start_day: date with format yy-mm-dd, if start date is none it is defaulted to the last monday
+            - end_day: date with format yy-mmd-dd, if end date is none it is defaulted to today
+    '''
+    start_date = request.args.get('start_date')
+    end_date = request.args.get('end_date')
+
+    if not start_date:
+        today = datetime.today().replace(hour=0, minute=0, second=0,
+                                         microsecond=0)
+        start_date = today - timedelta(days=today.weekday())
+    else:
+        start_date = datetime.strptime(start_date, '%Y-%m-%d')
+
+    if not end_date:
+        end_date = datetime.today().replace(hour=0, minute=0, second=0,
```

Comment:
I *think* it might be wise to do until this moment rather than until midnight. Otherwise, there's no easy way to get today's failures as well. What do you think?